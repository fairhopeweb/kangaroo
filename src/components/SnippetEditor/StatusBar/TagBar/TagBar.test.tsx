import { configure, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import renderer from 'react-test-renderer';

import TagBar from './TagBar';

configure({ adapter: new Adapter() });

describe('<TagBar />', () => {
  const mockProps = {
    tags: '1,2,3',
    onSelect: jest.fn(),
  };

  it('render without crashing', () => {
    shallow(<TagBar {...mockProps} />);
  });

  it('matches snapshot', () => {
    const wrapper = renderer.create(<TagBar {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
