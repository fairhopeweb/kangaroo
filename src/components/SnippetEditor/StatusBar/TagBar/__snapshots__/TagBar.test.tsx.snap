// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagBar /> matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="wrapper"
  >
    <div
      className="tag"
      onClick={[Function]}
      style={
        Object {
          "background": "#F55656",
        }
      }
    >
      <span
        aria-hidden={true}
        className="bp3-icon bp3-icon-small-cross actionIcon"
        icon="small-cross"
      >
        <svg
          data-icon="small-cross"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <path
            d="M9.41 8l2.29-2.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-2.29-2.3a1.003 1.003 0 00-1.42 1.42L6.59 8 4.3 10.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l2.29 2.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="wrapper"
  >
    <div
      className="tag"
      onClick={[Function]}
      style={
        Object {
          "background": "#F29D49",
        }
      }
    >
      <span
        aria-hidden={true}
        className="bp3-icon bp3-icon-small-cross actionIcon"
        icon="small-cross"
      >
        <svg
          data-icon="small-cross"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <path
            d="M9.41 8l2.29-2.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-2.29-2.3a1.003 1.003 0 00-1.42 1.42L6.59 8 4.3 10.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l2.29 2.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="wrapper"
  >
    <div
      className="tag"
      onClick={[Function]}
      style={
        Object {
          "background": "#F2B824",
        }
      }
    >
      <span
        aria-hidden={true}
        className="bp3-icon bp3-icon-small-cross actionIcon"
        icon="small-cross"
      >
        <svg
          data-icon="small-cross"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <path
            d="M9.41 8l2.29-2.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-2.29-2.3a1.003 1.003 0 00-1.42 1.42L6.59 8 4.3 10.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l2.29 2.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="wrapper"
  >
    <div
      className="tag"
      onClick={[Function]}
      style={
        Object {
          "background": "#15B371",
        }
      }
    >
      <span
        aria-hidden={true}
        className="bp3-icon bp3-icon-small-plus actionIcon"
        icon="small-plus"
      >
        <svg
          data-icon="small-plus"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <path
            d="M11 7H9V5c0-.55-.45-1-1-1s-1 .45-1 1v2H5c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1V9h2c.55 0 1-.45 1-1s-.45-1-1-1z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="wrapper"
  >
    <div
      className="tag"
      onClick={[Function]}
      style={
        Object {
          "background": "#2B95D6",
        }
      }
    >
      <span
        aria-hidden={true}
        className="bp3-icon bp3-icon-small-plus actionIcon"
        icon="small-plus"
      >
        <svg
          data-icon="small-plus"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <path
            d="M11 7H9V5c0-.55-.45-1-1-1s-1 .45-1 1v2H5c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1V9h2c.55 0 1-.45 1-1s-.45-1-1-1z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
